app/src/main/java/com/youssef/fixit/MainActivity/SplashScreen.java

```diff
@@ -24,16 +36,32 @@ protected void onCreate(Bundle savedInstanceState) {
             public void run() {
 
                 try {
-
                     sleep(3000);
-                    Intent intent = new Intent(SplashScreen.this, MainActivity.class);
-                    startActivity(intent);
-                    finish();
+                    CheckLogin();
                 } catch (InterruptedException e) {
                     e.printStackTrace();
                 }
             }
         };
         t.start();
     }
+
+    private void CheckLogin() {
+        MainActivity.MyToken = preferences.getString("token", "");
```

Comment:
This logic can be moved to the `SplashPresenter`, even if it seems like small logic it's always preferable to keep all the logic in the presenter layer to be testable. Also, in the future you code will expand and the logic will get bigger and more complicated so the presenter layer always a good resort